### PR TITLE
matcher_reverse_order_problem

### PR DESCRIPTION
**Pull request by koloale**

### CODE REVIEW

File: src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala [hunk 1]
```diff
@@ -14,7 +16,9 @@ case class MatcherSettings(enable: Boolean,
                            journalDataDir: String,
                            snapshotsDataDir: String,
                            snapshotsInterval: FiniteDuration,
-                           maxOpenOrders: Int
+                           maxOpenOrders: Int,
+                           baseAssets: Seq[String],
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala]: PriceAssets, PredefinedPairs -- more forex-like names.
+ change configs

File: src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala [hunk 2]
```diff
@@ -88,6 +88,8 @@ object LegacyConfigTransformer {
     "matcher.host"        -> "matcher.bind-address",
     "matcher.port"        -> "matcher.port",
     "matcher.minOrderFee" -> "matcher.order-match-tx-fee",
+    "matcher.baseAssets"  -> "matcher.base-assets",
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala]: We don't  have any PROD-running nodes with matchers, we do not need legacy-converters

File: src/main/scala/scorex/transaction/assets/exchange/Order.scala [hunk 3]
```diff
@@ -31,8 +54,8 @@ object OrderType {
 @SerialVersionUID(2455530529543215878L)
 case class Order(@ApiModelProperty(dataType = "java.lang.String") senderPublicKey: PublicKeyAccount,
                  @ApiModelProperty(dataType = "java.lang.String", example = "") matcherPublicKey: PublicKeyAccount,
-                 @ApiModelProperty(dataType = "java.lang.String") spendAssetId: Option[AssetId],
-                 @ApiModelProperty(dataType = "java.lang.String") receiveAssetId: Option[AssetId],
+                 assetPair: AssetPair,
```
Comment (ismagin) [inline src/main/scala/scorex/transaction/assets/exchange/Order.scala]: Does the missing annotation affect swagger sample?

File: waves-testnet.conf [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Waves node settins
 waves {
   # Node base directory
-  directory = ${HOME}"/waves"
+  directory = ${HOME}"/waves/testnet"
```
Comment (ismagin) [inline waves-testnet.conf]: do we need this?

File: waves-testnet.conf [hunk 5]
```diff
@@ -49,6 +49,16 @@ waves {
 
     # Matcher REST API port
     port = 6886
+
+    base-assets = [
```
Comment (ismagin) [inline waves-testnet.conf]: please change to ther predefs

File: src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala (hunk 2 above)
Comment (koloale) [inline src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala]: done

File: waves-testnet.conf (hunk 4 above)
Comment (koloale) [inline waves-testnet.conf]: done

File: waves-testnet.conf (hunk 5 above)
Comment (koloale) [inline waves-testnet.conf]: done

File: src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala (hunk 1 above)
Comment (koloale) [inline src/main/scala/com/wavesplatform/matcher/MatcherSettings.scala]: done

**Comment by koloale** [comment]
Review remarks has been fixed

File: src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala [hunk 6]
```diff
@@ -88,8 +88,6 @@ object LegacyConfigTransformer {
     "matcher.host"        -> "matcher.bind-address",
```
Comment (ismagin) [inline src/main/scala/com/wavesplatform/settings/LegacyConfigTransformer.scala]: Let's remove all matcher migrationssince it never got to prod yet.

**Pull request merged**